dev/archery/archery/linking.py

```diff
@@ -61,9 +64,71 @@ def list_dependency_names(self):
             names.append(name)
         return names
 
+    def _extract_symbols(self, symbol_info):
+        return [re.search(r'\S+$', line).group() for line in symbol_info if line]
```

Comment:
It seems that `lines` is better than `symbol_info` because you call it as `lines` in other places.

```suggestion
    def _extract_symbols(self, lines):
        return [re.search(r'\S+$', line).group() for line in lines if line]
```

Or we can use `symbol_info` in other places instead.